sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/UnivocityParser.scala

```diff
@@ -45,24 +45,41 @@ private[csv] class UnivocityParser(
   // A `ValueConverter` is responsible for converting the given value to a desired type.
   private type ValueConverter = String => Any
 
+  private val corruptFieldIndex = schema.getFieldIndex(options.columnNameOfCorruptRecord)
+  corruptFieldIndex.foreach { corrFieldIndex =>
+    require(schema(corrFieldIndex).dataType == StringType)
+    require(schema(corrFieldIndex).nullable)
+  }
+
+  private val inputSchema = StructType(schema.filter(_.name != options.columnNameOfCorruptRecord))
```

Comment:
It is because parsing CSV is dependent on the order of schema and tokens. In case of JSON, this can be just mapped by its key but for CSV it depends on the order of schema.

So, it seems this filters the corrupt field out in order to match the data schema with parsed tokens.